tests/topotests/bgp_extcomm_list_delete/test_bgp_extcomm-list_delete.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: ISC
+#
+# Copyright 2023 6WIND S.A.
+# Authored by Farid Mihoub <farid.mihoub@6wind.com>
+#
+
+"""
+bgp_extcomm_list-delete.py:
+
+Test the following commands:
+route-map test permit 10
+  set extended-comm-list <arg> delete
+"""
+
+import functools
+import json
+import os
+import pytest
+import re
+import sys
+
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+
+# pylint: disable=C0413
+from lib.topogen import Topogen, TopoRouter, get_topogen
+from lib.topolog import logger
+from lib import topotest
+
+
+pytestmark = [pytest.mark.bgpd]
+
+
+def build_topo(tgen):
+    for routern in range(1, 3):
+        tgen.add_router("r{}".format(routern))
+
+    switch = tgen.add_switch("s1")
+    switch.add_link(tgen.gears["r1"])
+    switch.add_link(tgen.gears["r2"])
+
+
+def setup_module(mod):
+    tgen = Topogen(build_topo, mod.__name__)
+    tgen.start_topology()
+
+    router_list = tgen.routers()
+
+    for i, (rname, router) in enumerate(router_list.items(), 1):
+        router.load_config(
+            TopoRouter.RD_ZEBRA, os.path.join(CWD, "{}/zebra.conf".format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_BGP, os.path.join(CWD, "{}/bgpd.conf".format(rname))
+        )
+
+    tgen.start_router()
+
+
+def teardown_module(mod):
+    tgen = get_topogen()
+    tgen.stop_topology()
+
+
+def test_bgp_maximum_prefix_invalid():
+    tgen = get_topogen()
+
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    r1 = tgen.gears["r1"]
+    r2 = tgen.gears["r2"]
+
+    def _bgp_converge():
+        output = json.loads(r2.vtysh_cmd("show ip bgp neighbor 192.168.255.1 json"))
+        expected = {
+            "192.168.255.1": {
+                "bgpState": "Established",
+                "addressFamilyInfo": {
+                    "ipv4Unicast": {
+                        "acceptedPrefixCounter": 2,
+                    }
+                },
+            }
+        }
+        return topotest.json_cmp(output, expected)
+
+    logger.info("*****Checking for bgp convergence")
+    test_func = functools.partial(_bgp_converge)
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=0.5)
+    assert result is None, "Can't converge initially"
+
+    def _configure_r1(ecom_t):
+        cmd = [
+            "con t\n",
+            "router bgp 65000\n",
+            "address-family ipv4 unicast\n",
+            f"redistribute connected route-map r2-out-{ecom_t}\n",
+            "exit-address-family\n",
+            "end",
+        ]
+        r1.vtysh_cmd("".join(cmd))
+
+    def _configure_r2(ecom_t, ecom):
+        cmd = f"con t\nbgp extcommunity-list standard r1 permit {ecom_t} {ecom}\n"
+        r2.vtysh_cmd(cmd)
+
+    def _bgp_extcomm_list_delete(ecom):
```

Comment:
This function is absolutely non-readable. Can we test each extended community separately (rt/soo/nt)? Like, assign rt to 10.10.10.1/32, soo 10.10.10.2/32, etc. Because now it's not clear in 15 seconds what's going on and what to expect.

Reply:
test updated.